trio/_socket.py

```diff
@@ -40,10 +38,17 @@
 
 T = TypeVar("T")
 
-# must use old-style typing because it's evaluated at runtime
-Address: TypeAlias = Union[
-    str, bytes, Tuple[str, int], Tuple[str, int, int], Tuple[str, int, int, int]
-]
+# _stdlib_socket.socket supports 13 different socket families, see
+# https://docs.python.org/3/library/socket.html#socket-families
+# and the return type of several methods in SocketType will depend on those. Typeshed
+# has ended up typing those return types as `Any` in most cases, but for users that
+# know which family/families they're working in we could make SocketType a generic type,
+# where you specify the return values you expect from those methods depending on the
+# protocol the socket will be handling.
+# But without the ability to default the value to `Any` it will be overly cumbersome for
+# most users, so currently we just specify it as `Any`.
+# AddressFormat = TypeVar("AddressFormat")
```

Comment:
Probably forgot to delete this.

Reply:
I kinda kept it in intentionally as what the line would otherwise look like. I could clarify the comment to make that more obvious, or delete the line as it's far from a very complicated one.

Reply:
clarified